AudioLoader/Audio_event_detection.py

```diff
@@ -0,0 +1,158 @@
+import torchaudio
+from pathlib import Path
+from typing import Tuple, Union
+from torch import Tensor
+import torch
+import os
+import time
+import json
+import tqdm
+import shutil
+import glob
+import multiprocessing as mp
+import warnings
+from distutils.dir_util import copy_tree
+from torchaudio.compliance import kaldi # for downsampling
+from torchaudio.datasets.utils import (
+    download_url,
+    extract_archive,
+)
+import hashlib
+import json
+import subprocess
+
+
+class DCASE2016(Dataset):
+    """Dataset class for DCASE2016 dataset.
+    Args:
+        root (str or Path): Path to the directory where the dataset is found or downloaded.
+            
+        split (str):
+            Choose different dataset splits such as ``"train"``, or ``"test"`` or ``"dev"``. (default: ``"train"``).
+        
+        download (bool, optional):
+            Whether to download the dataset if it is not found at root path. (default: ``False``).
+        
+    """
+    
+
+    def __init__(self,
+                 root: Union[str, Path],
+                 split: str = "train",                 
+                 download: bool = False
+                ):
+        self.split = split
+        if self.split == "train" or self.split =="dev":
+            url = "https://archive.org/compress/dcase2016_task2_train_dev/formats=ZIP&file=/dcase2016_task2_train_dev.zip"
+        else:
+            url = "https://archive.org/download/dcase2016_task2_test_public/dcase2016_task2_test_public.zip"
+        folder_name = 'DCASE2016'
+        
+        download_path = os.path.join(root, folder_name)
+        assert self.split.upper()=="TRAIN" or self.split.upper()=="TEST" or self.split.upper()=="DEV", f"split={split} is not present in this dataset"
+        
+        if self.split =="train":
+            archive_name = f'dcase2016_task2_train_dev.zip'
+            self._path = os.path.join(download_path, f"dcase2016_task2_train_dev",f"dcase2016_task2_train") 
+        elif self.split =="dev":
+            archive_name = f'dcase2016_task2_train_dev.zip'
+            self._path = os.path.join(download_path, f"dcase2016_task2_train_dev",f"dcase2016_task2_dev","sound") 
+        elif self.split == "test":
+            archive_name = f'dcase2016_task2_test_public.zip'
+            self._path = os.path.join(download_path, f"dcase2016_task2_test_public", "sound") 
+        
+        checksum_dict = {"test": "ac98768b39a08fc0c6c2ddd15a981dd7", "train":"0eab0635cff8e2d76e9e38b7a7de342d", "dev":"0eab0635cff8e2d76e9e38b7a7de342d"} #same value for train&dev
+
+        if download: #bad data format==> unzipped file has same name as zipped file, create tmp path and remove as a workaround
+            #file exists and extracted
+            if os.path.isfile(os.path.join(download_path,archive_name)) and os.path.exists(self._path):
+                print(f"Dataset archive exists, all files are extracted. Using all file from {self._path} ")
+            #file exists but not extracted
```

Comment:
The extraction is not automatic. The zip file contain nested zip files, this code does not deal with nested zip